include/deal.II/base/index_set.h

```diff
@@ -229,6 +229,12 @@ class IndexSet
   void compress () const;
 
   /**
+   * Reorder the ranges inside IndexSet so that the largest range is the first
+   * one.
+   */
+  void reorder_ranges();
```

Comment:
I agree that pushing this optimization down into the IndexSet class is a good idea. It's likely going to make some operations marginally slower and some unbelievably faster. What do you think of simply making it the default in `compress()`?